hbase-server/src/main/java/org/apache/hadoop/hbase/master/RegionStateStore.java

```diff
@@ -73,9 +73,11 @@
    * @return A ServerName instance or {@link HRegionInfo#getServerName(Result)}
    * if necessary fields not found or empty.
    */
-  static ServerName getRegionServer(final Result r, int replicaId) {
+  static ServerName getRegionServer(final Result r, int replicaId, Configuration config) {
     Cell cell = r.getColumnLatestCell(HConstants.CATALOG_FAMILY, getServerNameColumn(replicaId));
-    if (cell == null || cell.getValueLength() == 0) {
+    boolean isZKAssignmentInUse = ConfigUtil.useZKForAssignment(config) && !config
+      .getBoolean("hbase.assignment.usezk.migrating", false);
+    if (cell == null || cell.getValueLength() == 0 || isZKAssignmentInUse) {
```

Comment:
this is a change in logic from what was there before. Before, we would use `info:sn` if it was present, even while `usezk.migrating` is `true`. Are we saying it's better to change the behavior so we don't use it at all until `usezk` is set to `false`? And the purpose of the `migrating` flag is only to populate the `info:sn` and `info:state` entries, and not to use them at all? I just wanted to make sure I understood this change correctly.

Reply:
The intention is:
if (migrating to non-zk or enabled non-zk) => use `info:sn` if available, else use `info:server` as usual.
if (usezk) => never use `info:sn`, hence `isZKAssignmentInUse` is added with `||` condition so that if condition will always be true even if `info:sn` has non-empty value for ZK based assignment.